src/lib/format.js

```diff
@@ -24,14 +24,18 @@ export function valueFromContext(ctx, id, args) {
 export function entityFromContext(ctx, id, args) {
   const entity = ctx.messages.get(id);
 
-  if (!entity)  {
+  if (entity === undefined)  {
     return [
       { value: id, attrs: null },
       [new L10nError(`Unknown entity: ${id}`)]
     ];
   }
 
-  const [value] = ctx.format(entity, args);
+  let value = null;
+
+  if (typeof entity === 'string' || Array.isArray(entity) || entity.val !== undefined) {
+    value = ctx.format(entity, args)[0];
+  }
```

Comment:
I fixed this in https://github.com/l20n/l20n.js/commit/ed19609d1cda0fd4013c2a057c565bd3623e23fe by adding `MessageContext.formatToPrimitive` which can return string or null values.